services/shhext/service.go

```diff
@@ -184,8 +185,9 @@ type tracker struct {
 	w       *whisper.Whisper
 	handler EnvelopeEventsHandler
 
-	mu    sync.Mutex
-	cache map[common.Hash]EnvelopeState
+	mu      sync.Mutex
+	cache   map[common.Hash]EnvelopeState
+	batches map[common.Hash]map[common.Hash]struct{}
```

Comment:
I would call it something like `batchesWaitingConfirmation`.
Optionally, I would make a separate type for it with a specific limited api that we need.

Reply:
Why is it a map of maps and not a map of lists?

Reply:
we will receive multiple events with same batch and envelope, but a different peer id. it will be harder to filter out duplicates if it will be a list